lib/connection_controller.go

```diff
@@ -48,7 +41,12 @@ type ConnectionController struct {
 	// persistentIpToRemoteNodeIdsMap maps persistent IP addresses, like the --connect-ips, to the RemoteNodeIds of the
 	// corresponding RemoteNodes. This is used to ensure that we don't connect to the same persistent IP address twice.
 	// And that we can reconnect to the same persistent IP address if we disconnect from it.
-	persistentIpToRemoteNodeIdsMap map[string]RemoteNodeId
+	persistentIpToRemoteNodeIdsMap *collections.ConcurrentMap[string, RemoteNodeId]
+
+	validatorMapLock sync.RWMutex
+	// validatorMap is a list of all validators that we are connected to. It will be updated periodically by the
+	// owner of the ConnectionController.
+	validatorMap *collections.ConcurrentMap[bls.SerializedPublicKey, consensus.Validator]
```

Comment:
Please clarify the naming. If I understand correctly, these are the validators the NetworkManager wants to connect to
```suggestion
	validatorsToConnectLock sync.RWMutex
	// validatorsToConnect is a map of all validators registered in consensus that we need to connect to.
	// It will be updated periodically by the owner of the ConnectionController.
	validatorsToConnect *collections.ConcurrentMap[bls.SerializedPublicKey, consensus.Validator]
```